commons/com.b2international.index/src/com/b2international/index/revision/StagingArea.java

```diff
@@ -937,4 +937,15 @@ public StagedObject removed(Object object, RevisionDiff diff, boolean commit) {
 		return new StagedObject(StageKind.REMOVED, object, diff, commit);
 	}
 
+	public void removeExclusions(Set<String> exclusions) {
```

Comment:
It would be better to apply the exclusions immediately when we set them on the `StagingArea` and not relying on a method call with `removeExclusions`.
Also, this eliminates the actual objects from the `StagingArea` while I think it would be better to keep them around and just apply the exclusions as a filter on top of them.